src/Core/ObjectRegistry.php

```diff
@@ -110,30 +112,26 @@ protected function _checkDuplicate($name, $config)
     {
         $existing = $this->_loaded[$name];
         $msg = sprintf('The "%s" alias has already been loaded', $name);
-        $hasConfig = false;
-        if (method_exists($existing, 'config')) {
-            $hasConfig = true;
-        }
-        if (!$hasConfig) {
+        if (!method_exists($existing, 'config')) {
             throw new RuntimeException($msg);
         }
-        if (empty($config)) {
-            return;
-        }
+
```

Comment:
Now we had forced `$config` in `load()` must be an array, so this is unnecessary anymore.